### PR TITLE
fix: pass APIVersions value to manifest generation request during app validation and during app manifests loading

### PR DESCRIPTION
**Pull request by alexmt**
The https://github.com/argoproj/argo-cd/pull/3243 introduced support of `--api-versions` flag during helm chart rendering . The value for `--api-versions` flag is supplied by application controller but does not supply by API server while executing `Create` and `GetManifests` API. This PR fixes that issue.

### CODE REVIEW

File: util/argo/argo.go [hunk 1]
```diff
@@ -310,6 +316,17 @@ func ValidatePermissions(ctx context.Context, spec *argoappv1.ApplicationSpec, p
 	return conditions, nil
 }
 
+// APIGroupsToVersions converts list of API Groups into versions string list
+func APIGroupsToVersions(apiGroups []metav1.APIGroup) []string {
```
Comment (jannfis) [inline util/argo/argo.go]: Can we have a test for this method, please?

**Review by jannfis — CHANGES_REQUESTED** [review]
Looks good to me so far, just needs tests.

File: util/argo/argo.go (hunk 1 above)
Comment (alexmt) [reply util/argo/argo.go]: I knew you would ask. Sorry for being lazy. Added.

Comment (jannfis) [reply util/argo/argo.go]: Thanks :+1:

**Review by jannfis — APPROVED** [review]
LGTM

**Pull request merged**